### PR TITLE
BOOKKEEPER-997: Wire protocol change for supporting long poll

### PR DESCRIPTION
**Pull request by sijie**
adding fields in ReadRequest to support piggying back lac on read responses and support long poll requests.

### CODE REVIEW

**Comment by eolivelli** [comment]
@sijie 
Maybe it will be better to run protobuf compiler e commit new java classes in this commit, so that anyone who will run the compiler will not find inconsistent .java sources

**Comment by sijie** [comment]
@eolivelli that works for me. can you explore the solution for it?

**Comment by eolivelli** [comment]
@sijie 
I meant to run simply this command and "git add" the generated classes to the commit
`mvn clean install -Pprotobuf -DskipTests`

I can pick up your patch if you want

**Comment by sijie** [comment]
oh, I understand what you are saying now. The reason that we didn't make it automatically generate the protobuf classes with the build is not every one installed protoc. If there's a plugin can automatically install protoc, I am good with that. 

For this pull requests, I will manually generate the proto classes. Feel free to create a JIRA for automate this in the build.

**Comment by eolivelli** [comment]
I think that to generate  automatically sources is not useful and would keep the build less reproducible and slow.

I was saying that usually we commit the generated clasdses together with the change to the protocol file.
I image you are going to do it in another commit but it would ve clearer if we do it in thia commit.

**Comment by sijie** [comment]
@eolivelli okay. so I think we are in the same page. I will add the generated protobuf classes in this review. I forgot that.

**Comment by sijie** [comment]
add the generated protobuf file.

**Comment by sijie** [comment]
tag @merlimat for reviews

File: bookkeeper-server/src/main/proto/BookkeeperProtocol.proto [hunk 1]
```diff
@@ -85,13 +85,18 @@ message Request {
...
     }
     optional Flag flag = 100;
     required int64 ledgerId = 1;
     // entryId will be -1 for reading the LAST_ADD_CONFIRMED entry.
     required int64 entryId = 2;
     // Used while fencing a ledger.
     optional bytes masterKey = 3;
+    // Used for waiting on last add confirmed update
+    optional int64 previousLAC = 4;
+    // Used as a timeout for the long polling request
```
Comment (merlimat) [inline bookkeeper-server/src/main/proto/BookkeeperProtocol.proto]: Please add time unit for the timeout. Just a mention in the comment would be good

Comment (sijie) [inline bookkeeper-server/src/main/proto/BookkeeperProtocol.proto]: done.

**Comment by sijie** [comment]
ping @merlimat

**Pull request closed without merging**